external-plugins/rehearse/plugin/handler/handler.go

```diff
@@ -0,0 +1,504 @@
+func (h *gitHubEventsHandler) modifiedJobConfigs(changedFiles []string) ([]string, error) {
+	var absModifiedProwConfigs []string
+	for _, changedFile := range changedFiles {
+		if strings.HasPrefix(changedFile, h.jobsConfigBase) {
+			if strings.HasSuffix(changedFile, ".yaml") || strings.HasSuffix(changedFile, ".yml") {
+				log.Infof("A modified config found: %s", changedFile)
+				absModifiedProwConfigs = append(absModifiedProwConfigs, changedFile)
+			}
+		}
+		log.Debugf("Skipping file: %s. Not a Prow/Jobs config", changedFile)
+	}
+	return absModifiedProwConfigs, nil
+}
+
+func jobKeyFunc(repo string, presubmit config.JobBase) string {
+	return fmt.Sprintf("%s#%s", repo, presubmit.Name)
+}
+
+func repoFromJobKey(jobKey string) string {
+	s := strings.Split(jobKey, "#")
+	r := s[:1]
+	return strings.Join(r, "/")
+}
+
+func hashPeriodicsConfig(periodics []config.Periodic) map[string]config.Periodic {
+	p := map[string]config.Periodic{}
+	for _, periodic := range periodics {
+		p[periodic.JobBase.Name] = periodic
+	}
+	return p
+}
+
+func hashPresubmitsConfig(presubmits map[string][]config.Presubmit) map[string]config.Presubmit {
+	presubmitsFlat := map[string]config.Presubmit{}
+	for repo, presubmitsForRepo := range presubmits {
+		for _, presubmit := range presubmitsForRepo {
+			presubmitsFlat[jobKeyFunc(repo, presubmit.JobBase)] = presubmit
+		}
+	}
+	return presubmitsFlat
+}
+
+// catFile executes a git cat-file command in the specified git dir and returns bytes representation of the file
+func catFile(log *logrus.Logger, gitDir, file, refspec string) ([]byte, int) {
+	cmd := exec.Command("git", "-C", gitDir, "cat-file", "-p", fmt.Sprintf("%s:%s", refspec, file))
+	log.Debugf("Executing git command: %+v", cmd.Args)
+	out, _ := cmd.CombinedOutput()
+	return out, cmd.ProcessState.ExitCode()
+}
+
+func writeTempFile(log *logrus.Logger, basedir string, content []byte) (string, error) {
+	tmpfile, err := ioutil.TempFile(basedir, "job-config")
+	if err != nil {
+		log.WithError(err).Errorf("Could not create temp file for job config.")
+		return "", err
+	}
+	defer tmpfile.Close()
+	_, err = tmpfile.Write(content)
+	if err != nil {
+		log.WithError(err).Errorf("Could not write data to file: %s", tmpfile.Name())
+		return "", err
+	}
+	tmpfile.Sync()
+	return tmpfile.Name(), nil
+}
+
+func addExtraRefs(refs []prowapi.Refs, org, repo, ref string) []prowapi.Refs {
+	// We don't want to modify the refs on the original object, so we make our copy of it
+	r := reflect.ValueOf(make([]prowapi.Refs, len(refs)))
+	reflect.Copy(r, reflect.ValueOf(refs))
+
+	copiedRefs := r.Interface().([]prowapi.Refs)
+
+	if copiedRefs != nil {
+		// We need to check if there is an extra ref that is set to be the WorkDir.
+		// If there is no such ref, then we set the repo for which this presubmit is
+		// configured for as the work dir. Otherwise, we don't want to override the
+		// users configuration.
+		copiedRefs = append(
```

Comment:
You can keep just this and remove the surrounding if/else. You can iterate on `nil` arrays and `append` to nil arrays.
See this example: https://play.golang.org/p/gLkZP3vocO1

Reply:
I always tend to forget that iteration on nil arrays works in Go :laughing: 
Thanks.